impeller/entity/contents/solid_color_contents.cc

```diff
@@ -101,4 +101,23 @@ std::unique_ptr<SolidColorContents> SolidColorContents::Make(const Path& path,
   return contents;
 }
 
+std::optional<Color> SolidColorContents::AsBackgroundColor(
+    const Entity& entity,
+    ISize target_size) const {
+  if (!(GetColor().IsOpaque() &&
+        (entity.GetBlendMode() == BlendMode::kSource ||
+         entity.GetBlendMode() == BlendMode::kSourceOver))) {
+    return {};
+  }
+
+  std::optional<Rect> coverage = GetCoverage(entity);
+  if (!coverage.has_value()) {
+    return {};
+  }
+
+  Rect target_rect = Rect::MakeSize(target_size);
+  return coverage.value().Contains(target_rect) ? GetColor()
+                                                : std::optional<Color>();
```

Comment:
Checking if the `Entity` coverage covers the whole pass isn't enough to determine that the drawn geometry will cover all fragments in the subpass. Additional restrictions are needed.

For example, if the geometry is a stroke path that draws a line across the diagonal of the screen, then this check will incorrectly pass.

I'd suggest adding something like `virtual bool Geometry::CoversArea(const Entity& entity, Rect area) const` for determining if the given rect is completely covered by the geometry.
  * Default implementation just returns `false`.
  * Add an override for `CoverGeometry` which returns `true` to handle the `DrawPaint` case that we're already handling.
  * Add an override for `RectGeometry` to handle the new case we're trying to tackle.

For the `RectGeometry` case, checking against the geometry's coverage will work if the entity's transform is known to be translation+scale only (we've already got a helper that can be used to check for that).

Reply:
Oh thanks.  I wondered about this but somehow convinced myself we were only hitting this with rect geometry.  I'll create a new test for this case.

Reply:
Done